tools/run_tests/run_xds_tests.py

```diff
@@ -2554,6 +2554,7 @@ def create_url_map(gcp, name, backend_service, host_name):
         project=gcp.project, body=config).execute(num_retries=_GCP_API_RETRIES)
     wait_for_global_operation(gcp, result['name'])
     gcp.url_map = GcpResource(config['name'], result['targetLink'])
+    gcp.url_maps.append(gcp.url_map)
```

Comment:
I'd be a little wary of this keeping both `gcp.url_map` _and_ `gcp.url_maps`, because (as you point out) the global "single" `gcp.url_map` that will be used elsewhere will just end up being whatever url map is created or retrieved last (likewise with target proxies and global forwarding rules). I would suggest modifying this function to drop `gcp.url_map` altogether and instead looking something like what is done with `create_backend_service`:

```
url_map = GcpResource(config['name'], result['targetLink'])
gcp.url_maps.append(gcp.url_map)
return url_map
```

Reply:
Yea I saw your comment. :) I removed gcp.url_map and replace it with gcp.url_maps, same as targe_proxy and forwarding _rules, and allow specifying which dependency resource to use when creating a resource. And we have to be careful about the index because they have to be fully aligned.

Reply:
Previously there was a set of global `gcp.url_map` `gcp.target_proxy` and `gcp.forwarding_rule` resources that was referenced with each other. If we replace them with a list because we want to track them all, then the first one in the list will be the default one - for the test case that only care about one set global resources, and we allow specifying which dependency resource to be used - for the test cases that need multiple set of urlmap+tp+fr.
This is different from backend_services, because they don't get involved in the dependency chain nor do they need to be deleted for the test verification.
I actually feel like the previous version had its advantage but whatever.